NTP plugin update interval

barlima opened the pull request:
The plugin has been updated, you can see the result here https://server.pingdom.com/psm_dev/clients/10809861

Customers are now able to set the interval between `netstat` check to avoid unwanted behaviour (plugin failing).

One important thing - If you run a plugin and because of some `if` statements the `remember` function will not be called, the memory will forget all of the values.

andre commented on ntpstat/ntpstat.rb:
```diff
@@ -1,18 +1,47 @@
+
   def build_report
+    init_values if memory(:within).nil?
     output = `ntpstat`
```
The point is to only run `ntpstat` when necessary :) this runs it every time.

barlima commented:
I have modified the code and fixed some issues.
I hope it looks simpler now.

Bartek

On Fri, Jan 12, 2018 at 8:44 PM, Andre Lewis <notifications@github.com>
wrote:

> *@andre* commented on this pull request.
>
> The code looks more complicated than it needs to be. If it works, OK, but
> I think it could be simplified.
> ------------------------------
>
> In ntpstat/ntpstat.rb
> <https://github.com/pingdomserver/scout-plugins/pull/243#discussion_r161303336>
> :
>
> >    def build_report
> +    init_values if memory(:within).nil?
>      output = `ntpstat`
>
> The point is to only run ntpstat when necessary :) this runs it every
> time.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pingdomserver/scout-plugins/pull/243#pullrequestreview-88576060>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALT3m93N8Co-98ybD52tvkbO1BKpZeDqks5tJ7V4gaJpZM4RbAkM>
> .
>

barlima commented:
Hello,

I am sorry Andre, but I have made a mistake... I have update not the right
plugin :/ also NTP but not the one which the customer uses...
I have updated the correct plugin and the pull request available is here:
https://github.com/pingdomserver/scout-plugins/pull/245

Bartek

On Mon, Jan 15, 2018 at 7:24 PM, Andre Lewis <notifications@github.com>
wrote:

> *@andre* approved this pull request.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pingdomserver/scout-plugins/pull/243#pullrequestreview-88902615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALT3m31f4wYQnv-3UHokZHtKmQPSXSCaks5tK5fTgaJpZM4RbAkM>
> .
>
